### ISSUE TITLE
Failure to detect runtime-known argument passed to comptime parameter

### ISSUE THREAD

**Issue by wooster0** [ziglang/zig#16753]
### Zig Version

0.12.0-dev.25+36c57c3ba

### Steps to Reproduce and Observed Behavior

```zig
test {
    const x = struct {
        fn x(sentinel: u8) void {
            var buf: [64]u8 = undefined;
            _ = @import("std").mem.sliceTo(&buf, sentinel);
        }
    }.x;
    x(0);
}
```
This fails to detect that `sentinel` is a runtime-known argument and cannot be passed to comptime parameter `comptime end: meta.Elem(@TypeOf(ptr))` of `std.mem.sliceTo`:
```
$ zig test x.zig
Segmentation fault (core dumped)
```
It works when making `sentinel` a `comptime` parameter.

### Expected Behavior

```
$ zig test x.zig
x.zig:7:31: error: runtime-known argument passed to comptime parameter
            _ = @import("std").mem.sliceTo(&buf, sentinel);
                                                 ^~~~~~~~
x.zig:13:39: note: declared comptime here
pub fn sliceTo(ptr: anytype, comptime end: std.meta.Elem(@TypeOf(ptr))) SliceTo(@TypeOf(ptr), end) {
                                      ^~~
```

**Comment by mlugg** [comment]
Resolved by #16604

**Comment by wooster0** [comment]
Do you mind adding this as a test?
```zig
export fn entry() void {
    const x = struct {
        fn x(sentinel: u8) void {
            var buf: [64]u8 = undefined;
            _ = @import("std").mem.sliceTo(&buf, sentinel);
        }
    }.x;
    x(0);
}

// error
// backend=stage2
// target=native
//
// :5:50: error: runtime-known argument passed to comptime parameter
```

**Comment by mlugg** [comment]
Yeah, doing that now - just trying to reduce it to not depend on std

**Comment by jacobly0** [comment]
Already resolved by #16752 and already tested by `callComptimeBoolFunctionWithRuntimeBool`.
